app/models/user.rb

```diff
@@ -9,4 +9,55 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_arrayb = inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    friends_array.concat(friends_arrayb)
+    friends_array.compact
+  end
+
+  def send_invitation(user_id)
+    @friendship = Friendship.new(user_id: id, friend_id: user_id)
+    @friendship.confirmed = false
+    @friendship.save
+  end
+
+  # Users who have yet to confirme friend requests
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
+  end
```

Comment:
**Optional**. It is not a good practice to use Enumerable to retrieve objects from the database. Try changing this to an association instead. 
HINT: check all `friendships` and `inverse_friendships` where `confirmed: false`